pkg/topology/full/full.go

```diff
@@ -77,6 +77,14 @@ func (d *driver) AddPeer(ctx context.Context, addr swarm.Address) error {
 	if !isConnected(addr, connectedPeers) {
 		peerAddr, err := d.p2pService.Connect(ctx, ma)
 		if err != nil {
+			d.mtx.Lock()
+			delete(d.receivedPeers, addr.ByteString())
+			d.mtx.Unlock()
+			var e *p2p.ConnectionBackoffError
```

Comment:
since `e` is not really used to cast the error, wouldn't it be nicer to just remove this variable and check `errors.is(err, p2p.ConnectionBackoffError)`?

Reply:
Ah sorry, I did not sleep for 2 days, so I am a bit tired today 🙈 It needs to be asserted with .As, so the first code was correct I just need to switch `errors.Is` to `errors.As`.

Reply:
https://blog.golang.org/go1.13-errors